enhancements/microshift/lvms-topolvm-downstream-replacement.md

```diff
@@ -0,0 +1,269 @@
+  for using LVMS in MicroShift based on LVMCluster with an extended documentation
+  for specialised use cases.
+* Forcing MicroShift users to migrate to LVMS CustomResources. We will allow
+  the use of the old deviceClass mechanism without a migration path and permanently
+  allow custom deviceClass management.
+
+## Proposal
+
+With 4.16 and the [controller consolitation initative in LVMS](https://issues.redhat.com/browse/OCPEDGE-689)
+we no longer use any of our own upstream images like TopoLVM or the CSI Sidecars. 
+We would like to introduce a conformant way to rebase MicroShift based on LVMS, not TopoLVM in the future.
+This will allow us to no longer ship topolvm as a component,
+but embed it into LVMS fully as well as maintain LVMS as the only
+storage driver in MicroShift and OpenShift we have to support for lvm2 support.
+
+### Workflow Description
+
+1. Edge device administrator deploys a host with MicroShift 4.15
+   installed, and maintains a storageClass, and deviceClass as per 
+   [the documentation reference](https://github.com/openshift/microshift/blob/main/docs/contributor/storage/configuration.md).
+2. Software runs, time passes.
+3. Edge device administrator updates the host to run MicroShift 4.16.
+   * For ostree-based systems, the host is automatically rebooted as
+     part of the update process.
+   * For RPM-based systems, the user must reboot the host after the
+     software update is completed.
+4. Edge device restarts.
+5. MicroShift restarts.
+6. MicroShift removes the old TopoLVM deployment without removing the CRDs.
+   All existing storageClasses and deviceClasses are still present.
+7. MicroShift installs LVMS with a LVMCluster CustomResource.
+   * The LVMCluster is configured with `.spec.configurationPolicy=ExternalConfiguration`
+     which will trigger a deployment of the driver which can use the existing
+     deviceClasses and storageClasses.
+   * If LVMS does not detect a lvmd.yaml file (at runtime), it will error out,
+     as an external configuration is now mandatory for correct configuration.
+     Note that if the lvmd.yaml file is not present at install time, we should
+     keep the current behavior of not installing LVMS at all.
+   * When External Configuration is not used, then not specifying any deviceClasses
+     will result in rejection of the LVMCluster specification.
+   * Note that while it is eventually planned to decouple the CSI stack from MicroShifts
+     core resources, this enhancement only targets the migration of the existing solution,
+     not the adoption of LVMS as an external component out-of-tree.
+8. MicroShift continues to run, and LVMS starts the vg-manager DaemonSet with all CSI bindings included.
+9. All existing configurations continue to work
+   * Errors of TopoLVM in the node will now be written to vg-manager.
+   * Errors of the LVMCluster deployment will be written to the lvms-controller.
```

Comment:
Are these names of LVMS pods? Can we clarify in the text not to assume knowledge of LVMS internals?

Reply:
Now made it clear that we are working with 2 pods and that additional events get published. Also made clear that we are talking about a CustomResource as well as 2 pod logs.